Extension/src/LanguageServer/client.ts

```diff
@@ -1686,8 +1687,13 @@ export class DefaultClient implements Client {
                 if (changedSettings["legacyCompilerArgsBehavior"]) {
                     this.configuration.handleConfigurationChange();
                 }
-                if (changedSettings["default.compilerPath"] !== undefined || changedSettings["default.compileCommands"] !== undefined || changedSettings["default.configurationProvider"] !== undefined) {
-                    ui.ShowConfigureIntelliSenseButton(false, this);
+                const showButtonSender: string = "settingsChanged";
+                if (changedSettings["default.configurationProvider"] !== undefined) {
+                    ui.ShowConfigureIntelliSenseButton(false, this, ConfigurationType.ConfigProvider, showButtonSender);
```

Comment:
The auto-config type doesn't appear to be checked here, i.e. I see it send ConfgProvider instead of AutoConfigProvider after clearing out the setting.

Reply:
I think one of the primary goals for this telemetry is to identify the users who have figured out how to configure their stuff.  If we fired ConfigProvider in the past for the user because this default.configurationProvider setting had a value, then we will have already categorized this user as someone who knows how to configure themselves and it wouldn't really matter if another ConfigProvider type was sent for them.  So to avoid churning the code too much before we release, I'd like to ignore this one.

Reply:
sure